src/api/api.cpp

```diff
@@ -68,26 +68,39 @@ static rgl_status_t updateAPIState(rgl_status_t status, std::optional<std::strin
 		else {
 			RGL_CRITICAL("Unrecoverable error (code={}): {}", lastStatusCode, msg);
 		}
-		Logger::instance().flush();
+		Logger::getOrCreate().flush();
 	}
 	return status;
 }
 
+static void rgl_lazy_init()
+{
+	// Trigger initialization on the first API call
+	static bool firstCall = true;
+	if (firstCall && RGL_TAPE_FORCE) {
+		firstCall = false; // Set to false before calling to prevent infinite recursion
+		rgl_tape_record_begin(RGL_TAPE_PATH); // Calls rglSafeCall to handle exceptions
+		// If rgl_tape_record_begin(...) failed, it will turn into RGL_UNRECOVERABLE_ERROR after continuing rglSafeCall.
+	}
+	static auto& _logger = Logger::getOrCreate();
+	static auto& _optix = Optix::getOrCreate();
```

Comment:
micro-optimization:
```suggestion
	if (!firstCall {
	    return;
	}
	static auto& _logger = Logger::getOrCreate();
	static auto& _optix = Optix::getOrCreate();
	firstCall = false; // Set to false before RGL_TAPE_FORCE to prevent infinite recursion
	if (RGL_TAPE_FORCE) {
		rgl_tape_record_begin(RGL_TAPE_PATH); // Calls rglSafeCall to handle exceptions
		// If rgl_tape_record_begin(...) failed, it will turn into RGL_UNRECOVERABLE_ERROR after continuing rglSafeCall.
	}
```